Jenkinsfile

```diff
@@ -37,43 +23,54 @@ pipeline {
-            def slackURL = env.SLACK_URL
-            def text = (
-              "*<http://localhost:8080/job/${env.JOB_NAME}/${env.BUILD_NUMBER}|#${env.BUILD_NUMBER}>* *${env.JOB_NAME}* " +
-              "<https://bldr.habitat.sh/#/pkgs/${packageIdent}|${packageIdent}>\n" +
-              "<https://github.com/mozilla/reticulum/commit/$gitSha|$gitSha> " +
-              "Reticulum -> ${retPoolIcon} `${retPool}`: ```${gitSha} ${gitMessage}```\n" +
-              "<https://smoke-hubs.mozilla.com/0zuesf6c6mf/smoke-test?required_ret_version=${retVersion}&required_ret_pool=${retPool}|Smoke Test> - to push:\n" +
-              "`/mr ret deploy ${retVersion} ${retPool}`"
-            )
-            def payload = 'payload=' + JsonOutput.toJson([
-              text      : text,
-              channel   : "#mr-builds",
-              username  : "buildbot",
-              icon_emoji: ":gift:"
-            ])
-            sh "sudo /usr/bin/hab-pkg-promote '${packageIdent}' '${retPool}'"
-            sh "sudo /usr/bin/hab-pkg-promote '${packageIdent}' 'stable'"
-            sh "curl -X POST --data-urlencode ${shellString(payload)} ${slackURL}"
+            if (onlyPromoteToStage == "") {
+              def retPool = sh(returnStdout: true, script: "curl https://${poolHost}/api/v1/meta | jq -r '.pool'").trim()
+              sh "sudo /usr/bin/hab-pkg-promote '${packageIdent}' '${retPool}'"
+              sh "sudo /usr/bin/hab-pkg-promote '${packageIdent}' 'stable'"
+
+              def retPoolIcon = retPool == 'earth' ? ':earth_americas:' : ':new_moon:'
+              def gitMessage = sh(returnStdout: true, script: "git log -n 1 --pretty=format:'[%an] %s'").trim()
+              def gitSha = sh(returnStdout: true, script: "git log -n 1 --pretty=format:'%h'").trim()
+              def text = (
+                "*<http://localhost:8080/job/${jobName}/${buildNumber}|#${buildNumber}>* *${jobName}* " +
+                "<https://bldr.habitat.sh/#/pkgs/${packageIdent}|${packageIdent}>\n" +
+                "<https://github.com/mozilla/reticulum/commit/$gitSha|$gitSha> " +
+                "Reticulum -> ${retPoolIcon} `${retPool}`: ```${gitSha} ${gitMessage}```\n" +
+                "<https://smoke-hubs.mozilla.com/0zuesf6c6mf/smoke-test?required_ret_version=${retVersion}&required_ret_pool=${retPool}|Smoke Test> - to push:\n" +
+                "`/mr ret deploy ${retVersion} ${retPool}`"
+              )
+              sendSlackMessage(text, "#mr-builds", ":gift:", slackURL);
+            }
 
             // Upload to ret depot after publishing to slack to minimize wait
             sh 'sudo /usr/bin/hab-ret-pkg-upload $(ls -rt results/*.hart | head -n 1)'
+
+            if (onlyPromoteToStage == "true") {
+              sh "sudo /usr/bin/hab-ret-pkg-promote '${packageIdent}' '${stageChannel}'"
+
+              def text = "*${jobName}* promoted ${retVersion} to ${stageChannel}"
```

Comment:
It may be helpful to include some of the info / links that we do for the other case as well.
Something like
```
def text = "*${jobName}* promoted ${retVersion} to ${stageChannel}" + 
                "\nJob: *<http://localhost:8080/job/${jobName}/${buildNumber}|#${buildNumber}>* *${jobName}* " +
                "\nPackage: <https://bldr.habitat.sh/#/pkgs/${packageIdent}|${packageIdent}>\n" +
                "\nGit: <https://github.com/mozilla/reticulum/commit/$gitSha|$gitSha> " +
                "\nHead: ```${gitSha} ${gitMessage}```\n"
```